### PR TITLE
feat-724 added admin services and coressponding interfaces

### PR DESCRIPTION
**Pull request by danielsh28**

### CODE REVIEW

File: public/locales/en/translation.json [hunk 1]
```diff
@@ -86,7 +86,9 @@
   "header": {
     "Search": "Search",
     "User Greeting": "Hello",
-    "User Info Update": "Update Info"
+    "User Info Update": "Update Info",
+    "management" : "Management"
+
```
Comment (yuvalbl) [inline public/locales/en/translation.json:91]: remove space

File: public/locales/en/translation.json [hunk 2]
```diff
@@ -96,6 +98,9 @@
     "Email": "Email",
     "Organization": "Organization"
   },
+  "usersManagement": {
+  "title" : "Users Management"
```
Comment (yuvalbl) [inline public/locales/en/translation.json:102]: indentation

File: src/components/molecules/UserManagementDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import  React from 'react'
+import { IAnywayUserDetails } from '../../services/user.service';
+import DialogWithHeader from './DialogWithHeader';
+import { useTranslation } from 'react-i18next';
+
+
```
Comment (yuvalbl) [inline src/components/molecules/UserManagementDialog.tsx:6]: remove space

File: src/components/molecules/UserManagementDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+interface IUserManagementDialog {
+  users : Array<IAnywayUserDetails>;
+  isShowing : boolean;
+  onClose : () => any;
+}
+
+
```
Comment (yuvalbl) [inline src/components/molecules/UserManagementDialog.tsx:14]: remove space

File: src/components/molecules/UserManagementDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+const UserManagementDialog : React.FC<IUserManagementDialog> = ({users,isShowing, onClose}) => {
+  const { t } =  useTranslation();
+  return (
+    <DialogWithHeader title={t('usersManagement.title')} isShowing={isShowing} onClose={onClose}/>
+  )
+}
+
```
Comment (yuvalbl) [inline src/components/molecules/UserManagementDialog.tsx:21]: remove space

File: src/models/user/IUserRole.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+
+export default  interface IUserRole {
+  id: number;
+  name: string;
+  description: string
+}
+
```
Comment (yuvalbl) [inline src/models/user/IUserRole.ts:7]: remove space

File: src/components/molecules/UserProfileHeader.tsx [hunk 7]
```diff
@@ -30,33 +35,92 @@ const useStyles = makeStyles((theme) => ({
...
+}) => {
   const { t } = useTranslation();
-  const userData = userDetails.data;
+  const [loading, setLoading] = useState(false);
+  const store: RootStore = useStore();
+  const defaultFormInput: IFormInput = {
+    email: userDetails.data.email,
+    firstName: userDetails.data.firstName,
+    lastName: userDetails.data.lastName,
+    workplace: userDetails.data.roles.filter((role) => role !== ROLE_ADMIN_NAME)[0], // first role that is not admin
```
Comment (yuvalbl) [inline src/components/molecules/UserProfileHeader.tsx:56]: This line is not safe (array can be empty)

File: src/components/molecules/UserProfileHeader.tsx [hunk 8]
```diff
@@ -30,33 +35,92 @@ const useStyles = makeStyles((theme) => ({
...
+  isAdmin,
+}) => {
   const { t } = useTranslation();
-  const userData = userDetails.data;
+  const [loading, setLoading] = useState(false);
+  const store: RootStore = useStore();
+  const defaultFormInput: IFormInput = {
+    email: userDetails.data.email,
+    firstName: userDetails.data.firstName,
+    lastName: userDetails.data.lastName,
```
Comment (yuvalbl) [inline src/components/molecules/UserProfileHeader.tsx:55]: Not a must, but can be nice:

```suggestion
  const { email, firstName, lastName} = userDetails.data
  const defaultFormInput: IFormInput = {
    email,
    firstName,
    lastName,
```

File: src/components/molecules/UserProfileHeader.tsx [hunk 9]
```diff
@@ -30,33 +35,92 @@ const useStyles = makeStyles((theme) => ({
...
+  };
+  const saveUserChanges = async (email: string, prevOrgName: string, newOrgName: string) => {
+    if (newOrgName !== prevOrgName) {
+      try {
+        setLoading(true);
+        await store.setOrgToUser(newOrgName, email);
+        store.getUsersListInfo();
+      } catch (err) {
+        setLoading(false);
+        console.log(err);
```
Comment (yuvalbl) [inline src/components/molecules/UserProfileHeader.tsx:66]: remove log

File: src/components/molecules/UserProfileHeader.tsx [hunk 10]
```diff
@@ -30,33 +35,92 @@ const useStyles = makeStyles((theme) => ({
...
+    workplace: userDetails.data.roles.filter((role) => role !== ROLE_ADMIN_NAME)[0], // first role that is not admin
+  };
+  const saveUserChanges = async (email: string, prevOrgName: string, newOrgName: string) => {
+    if (newOrgName !== prevOrgName) {
+      try {
+        setLoading(true);
+        await store.setOrgToUser(newOrgName, email);
+        store.getUsersListInfo();
+      } catch (err) {
+        setLoading(false);
```
Comment (yuvalbl) [inline src/components/molecules/UserProfileHeader.tsx:65]: use `finally` to remove loader (should be turned off on success as well)

File: src/components/molecules/UserProfileHeader.tsx [hunk 11]
```diff
@@ -30,33 +35,92 @@ const useStyles = makeStyles((theme) => ({
...
+    }
+  }, [store]);
 
   return (
     <>
+      {true && (
+        <Box className={classes.userButton} onClick={() => toggleAdminManagementScreen(true)}>
+          {t('header.management')}
+        </Box>
+      )}
```
Comment (yuvalbl) [inline src/components/molecules/UserProfileHeader.tsx:94]: `true`?!??

File: src/components/molecules/UserProfileHeader.tsx [hunk 12]
```diff
@@ -30,33 +35,92 @@ const useStyles = makeStyles((theme) => ({
...
         {t('header.User Info Update')}
       </Box>
       <Box className={classes.welcomeMsg}>
-        <Typography.Body2>{`${t('header.User Greeting')} ${userData.firstName || ''}`}</Typography.Body2>
+        <Typography.Body2>{`${t('header.User Greeting')} ${defaultFormInput.firstName || ''}`}</Typography.Body2>
       </Box>
-      <Avatar className={classes.avatar} alt={userData.firstName?.substr(0, 1).toUpperCase()} src={userData.imgUrl} />
+      <Avatar
+        className={classes.avatar}
+        alt={defaultFormInput.firstName?.substr(0, 1).toUpperCase()}
```
Comment (yuvalbl) [inline src/components/molecules/UserProfileHeader.tsx:106]: why `.substr(0, 1)`?

File: src/components/molecules/adminManagement/AdminManagementForm.tsx [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+import { IusersForUpdateMap, IProps, labels, ISingleOrgDetails } from './types';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  table: {
+    border: `1px solid ${blackColor}`,
+    borderBottom: 0,
+    borderCollapse: 'separate',
+  },
+  tableContainer: {
+    maxHeight: 450,
```
Comment (yuvalbl) [inline src/components/molecules/adminManagement/AdminManagementForm.tsx:33]: use consts from global file

File: src/components/molecules/adminManagement/OrganizationEditList.tsx [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+        <Select   MenuProps={{
+          anchorOrigin: {
+            vertical: "bottom",
+            horizontal: "left"
+          },
+          getContentAnchorEl: null
+        }}
+                  name={itemEmail} value={currentOrganization} label="Organization" onChange={handleChange}>
+          {organizationsList &&
+            organizationsList.map((org: any) => {
```
Comment (yuvalbl) [inline src/components/molecules/adminManagement/OrganizationEditList.tsx:36]: why "any"?

File: src/components/molecules/adminManagement/types.tsx [hunk 15]
```diff
@@ -0,0 +1,32 @@
+export const labels = ['Name', 'Email', 'Organization'];
+
+export interface IProps {
+  isShowing: boolean;
+  onClose: () => void;
+  saveUserChanges: any;
```
Comment (yuvalbl) [inline src/components/molecules/adminManagement/types.tsx:6]: why any?

File: src/utils/utils.ts [hunk 16]
```diff
@@ -26,13 +26,19 @@ export const loginPopUpDim = {
 };
 //server url can be on dev or the heroku server
 export const serverUrl = process.env.REACT_APP_BASE_URL;
-
+export const ROLE_ADMIN_NAME = 'admins';
 export const authServerUrl = process.env.REACT_APP_AUTH_URL;
```
Comment (yuvalbl) [inline src/utils/utils.ts:30]:
```suggestion
export const AUTH_URL = process.env.REACT_APP_AUTH_URL;
```

**Review by yuvalbl — APPROVED** [review]

File: src/components/molecules/UserProfileHeader.tsx (hunk 10 above)
Comment (Harduf-l) [reply src/components/molecules/UserProfileHeader.tsx:65]: @yuvalbl when using "finally", the loading screen stops before the role was actually changed - it takes another extra millisecond to update the screen. that's why I used useEffect to detect when the role was actually changed, and then I change setLoading to false. 

  useEffect(() => {
    setLoading(false);
  }, [store.usersManagementTableData]);
  
  is that acceptable?

**Pull request merged**